src/shared/copy.c

```diff
@@ -798,6 +798,10 @@ static int fd_copy_regular(
         (void) futimens(fdt, (struct timespec[]) { st->st_atim, st->st_mtim });
         (void) copy_xattr(fdf, NULL, fdt, NULL, copy_flags);
 
+        r = fd_verify_regular_full(fdf, /* check_removed = */ true);
```

Comment:
tbh i think this should be a new helper, unrelated from verify_regular, after all an inode won't change its type half-way, hence we don't really need to check the type again which however the primary purpose of fd_verify_regular() is.

maybe instead add fd_verify_linked() that just returns 0 or -EDRM, and nothing else?